adapters/eea-direct.js

```diff
@@ -0,0 +1,265 @@
+  parallel([
+    (done) => {
+      map(records, (record, cb) => {
+        let measurement = {
+          coordinates: makeCoordinates(coordinates, record[11]),
+          parameter: record[5].toLowerCase(),
+          date: makeDate(record[16], record[4]),
+          value: parseInt(record[19]),
+          unit: record[record.length - 1],
+          attribution: [{
+            name: 'EEA',
+            url: source.sourceUrl
+          }],
+          averagingPeriod: {
+            unit: 'hours',
+            value: 1
+          }
+        };
+        // apply unit conversion to generated record
+        cb(null, convertUnits([measurement])[0]);
+      }, (err, mappedRecords) => {
+        if (err) {
+          return done(null, []);
+        }
+        done(null, mappedRecords);
+      });
+    },
+    (done) => {
+      // try to combine locations with coordinates
+      // by matching stationId or a spatial join.
+      // then return successful combos
+      request.get({url: stationsLink}, (err, res, locations) => {
+        if (err) {
+          return done(null, []);
+        }
+        locations = JSON.parse(locations);
+        map(coordinates, (crds, cb) => {
+          if (err) {
+            return cb(null, crds);
+          }
+          let geocodedProps = locations.find((location) => {
+            // an object with city, region, bounds, and stationId
+            // or just a stationId if reverse geocoding was unsuccessful
+            if (location.stationId) {
+              return location.stationId === crds.stationId;
+            } else {
+              return location === crds.stationId;
+            }
+          });
+          // if new station, denoted by no matching id,
+          // try to spatial join
+          if (!(geocodedProps)) {
+            geocodedProps = locations.find((location) => {
+              if (location.stationId) {
+                const bbox = bboxPolygon(location.bounds);
+                const stationPoint = point(
+                  crds.longitude,
+                  crds.latitude
+                );
+                return inside(stationPoint, bbox);
+              }
+            });
+          }
+          if (geocodedProps) {
+            geocodedProps = typeof geocodedProps === 'string' ? crds : Object.assign(geocodedProps, crds);
+            cb(null, geocodedProps);
+          }
+        }, (err, updatedCoordinates) => {
+          if (err) {
+            return done(null, coordinates);
+          }
+          done(null, updatedCoordinates);
+        });
+      });
+    }
+  ], (err, mappedData) => {
+    if (err) {
+      return cb(err, []);
+    }
+    let measurements = mappedData[0];
+    const coordinates = mappedData[1];
+    // map coordinates and location names to measurements
+    map(measurements, (measurement, done) => {
+      let station = matchStation(coordinates, measurement['coordinates']);
+      measurement['location'] = (station.location ? station.location : 'unused');
```

Comment:
Please don't use `unused`, if we have no valid `location` or `city` name, just leave it undefined so it'll fail an insert or better yet, just exclude it from the list to be saved.